codegen/smithy-ruby-codegen/src/main/java/software/amazon/smithy/ruby/codegen/middleware/factories/AuthMiddlewareFactory.java

```diff
@@ -44,22 +40,27 @@ public static Middleware build(GenerationContext context) {
         Map<ShapeId, Trait> serviceAuthSchemes =
                 ServiceIndex.of(context.model()).getAuthSchemes(context.service());
 
+        // get additional auth schemes
```

Comment:
Should this be done once and set on the context?

Reply:
Ideally. But I also need a way to link it up to the model because I need to extract properties from the traits...